lib/kafka/produce_operation.rb

```diff
@@ -137,6 +137,9 @@ def handle_response(response)
               delay: ack_time - message.create_time,
             })
           end
+        rescue Kafka::MessageSizeTooLarge
+          @logger.error "Message is too large when writing to #{topic}/#{partition}"
+          @buffer.clear_messages(topic: topic, partition: partition)
```

Comment:
I'm not 100% sure what the behavior is on the broker side when one of the messages in a batch is too large (or is corrupt) – do the rest of the messages get appended to the partition, only those up until the corrupt one, or none of them?

It's only safe to clear the buffer when the broker has acknowledged that the messages have all been appended, otherwise we're losing data. Whether we can assume that the only message left unappended will never succeed anyway and can therefore be cleared depends on how the brokers handle this situation. We'd need to look into that.

Note that I think the CorruptMessage behavior in ruby-kafka is wrong – we shouldn't just log when that happens and then clear the buffer afterwards, I'd prefer it if we crash.